src/extension/background-script/actions/ln/__tests__/request.test.ts

```diff
@@ -0,0 +1,326 @@
+import utils from "~/common/lib/utils";
+import type Connector from "~/extension/background-script/connectors/connector.interface";
+import db from "~/extension/background-script/db";
+import type { MessageGenericRequest, OriginData } from "~/types";
+
+import request from "../request";
+
+// suppress console logs when running tests
+console.error = jest.fn();
+
+jest.mock("~/common/lib/utils", () => ({
+  openPrompt: jest.fn(() => Promise.resolve({ data: {} })),
+}));
+
+// overwrite "connector" in tests later
+let connector: Connector;
+const ConnectorClass = jest.fn().mockImplementation(() => {
+  return connector;
+});
+
+jest.mock("~/extension/background-script/state", () => ({
+  getState: () => ({
+    getConnector: jest.fn(() => Promise.resolve(new ConnectorClass())),
+  }),
+}));
+
+const allowanceInDB = {
```

Comment:
We don't type this because the function will complain if it's the wrong type?

Reply:
If I get you correctly:
Both the allowance and permission are typed.
If I would remove a key/value from one of them TS will complain